scripts/chart/build

```diff
@@ -14,3 +14,17 @@ cp -rf ${1} build/chart/rancher
 
 sed -i -e "s/%VERSION%/${CHART_VERSION}/g" build/chart/rancher/Chart.yaml
 sed -i -e "s/%APP_VERSION%/${VERSION}/g" build/chart/rancher/Chart.yaml
+
+post_delete_base=$(grep -i shell-image pkg/settings/setting.go | cut -d "," -f 2 | sed -e 's/"//g' | sed -e 's/)//g' | sed -e 's/ //g') || ""
```

Comment:
I see what you are doing here, but I'm really not in favor of running sed through Go code to infer the base image of this chart. IMO, we should be just hard-coding the base image in the chart instead of relying on build magic.

The Version / AppVersion are different since they are small sed commands, but this one is really hard to parse through (because parsing a Go file is not something bash scripts can / should do). This is also relatively fragile since it forces us to never add something with the word `shell-image` before [this](https://github.com/rancher/rancher/blob/b0d3575cc216c5191590b39f23f2888f4e731a69/pkg/settings/setting.go#L97) in our Go code, which is a weird constraint to have on our code.

Was this a strict requirement for this issue?

Reply:
No it was just best solution we came up with, grepping for the image. There's no easy way to get to it from this point in build process. However I do like idea of just hardcoding the image name and then parsing for image version. We had thought of that but went this route so that if imaged changed it would catch it, but you are right its very fragile.

Reply:
the PR is updated. Now the image name is hardcoded, and only the image tag is grepped from the go file.

Reply:
If we are anyways grepping the image tag, we're imposing a constraint on the Go code regardless of whether we get the image name or not. So IMO we might as well go back to grepping the image name itself as well; my initial preference was to remove this sed script entirely in favor of hard-coding both the image and image tag.

However, I do have one idea: since rancher/rancher is using [Go 1.16](https://github.com/rancher/rancher/blob/baa461856167e28afea67c12667e92b3f242f467/go.mod#L3) which supports [`go:embed`](https://blog.carlmjohnson.net/post/2021/how-to-use-go-embed/), maybe the solution here is to set

```go
const (
        //go:embed shell_image.txt
	ShellDockerImage string
        //go:embed shell_image_tag.txt
        ShellDockerTag string
)

var (
// ...
	ShellImage                        = NewSetting(ShellDockerImage, ShellDockerTag)
// ...
)
```

That way we can just `cat pkg/settings/shell_image.txt` and `cat pkg/settings/shell_image_tag.txt` to get the image and tag for this script instead of sedding.

Otherwise, let's go back to grepping both the image tag + image name.

Any thoughts @cbron @jiaqiluo?

Reply:
We grep the image name and tag from `pkg/settings/setting.go` because we want to avoid inconsistency and extra dependencies. Using `go:embed` means we will have 2 more files to maintain which is not an ideal solution. 

Going back to grepping both the image tag + image name looks like an acceptable workaround to me.

Reply:
Agreed

Reply:
Thats very interesting solution Arvind. We could also just use a const there with a string to make grepping settings.go easier. But I am good with this solution for now.